overlord/ifacestate/handlers.go

```diff
@@ -195,25 +199,29 @@ func (m *InterfaceManager) setupProfilesForSnap(task *state.Task, _ *tomb.Tomb,
 		affectedSnaps = append(affectedSnaps, snapInfo)
 		confinementOpts = append(confinementOpts, confinementOptions(snapst.Flags))
 	}
-	return m.setupSecurityByBackend(task, affectedSnaps, confinementOpts)
+
+	return m.setupSecurityByBackend(task, affectedSnaps, confinementOpts, tm)
 }
 
 func (m *InterfaceManager) doRemoveProfiles(task *state.Task, tomb *tomb.Tomb) error {
 	st := task.State()
 	st.Lock()
 	defer st.Unlock()
 
+	perfTimings := timings.NewForTask(task)
```

Comment:
I didn't think of this so far, but should we capture task status to be able to distinguish doing from undoing?

Reply:
Done.